### PR TITLE
Create knative/build controller skeleton

### PR DESCRIPTION
**Pull request by fejta**
/assign @stevekuznetsov @BenTheElder @cjwagner 

Extension of https://github.com/kubernetes/test-infra/pull/8866 (which just vendors knative/build) that creates a controller.

All the code in `prow/cmd/build/controller.go` has been elided for the moment to simplify review of the next PR, which will use all these imports.

This code primarily just imports all the standard kubernetes apimachinery.

Thanks Steve for the help getting this working!

### CODE REVIEW

File: prow/apis/prowjobs/v1/types.go [hunk 1]
```diff
@@ -127,6 +130,11 @@ type ProwJobSpec struct {
 	// a Kubernetes agent
 	PodSpec *corev1.PodSpec `json:"pod_spec,omitempty"`
 
+	// BuildSpec provides the basis for running the test as
+	// a build-crd resource
+	// https://github.com/knative/build
+	BuildSpec *buildv1alpha1.BuildSpec `json:"build_spec,omitempty"`
```
Comment (fejta) [inline prow/apis/prowjobs/v1/types.go]: This is the change from #8866

File: prow/cmd/build/controller.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	_ "k8s.io/client-go/tools/cache"
+	_ "k8s.io/client-go/tools/clientcmd"
+	_ "k8s.io/client-go/tools/record"
+	_ "k8s.io/client-go/util/workqueue"
+
+	"github.com/sirupsen/logrus"
+)
+
+func main() {
+	logrus.Fatalf("see next pr")
```
Comment (fejta) [inline prow/cmd/build/controller.go]: This is the code change in this PR. Everything else is from #8866 or auto-generated as a result of the above imports.

Comment (BenTheElder) [reply prow/cmd/build/controller.go]: heh, nice

**Comment by BenTheElder** [comment]
/woof
for all that vendor

**Comment by stevekuznetsov** [comment]
That is one adorable woofer <3

**Pull request merged**